src/compression.rs

```diff
@@ -134,6 +134,7 @@ mod test {
         }
     }
 
+    #[allow(clippy::vec_init_then_push)]
     fn methods() -> Vec<CompressionMethod> {
```

Comment:
I think clippy wants code like this, which is an alright improvement.
```rust
vec![
    CompressionMethod::Stored
    #[cfg(any(
        feature = "deflate",
        feature = "deflate-miniz",
        feature = "deflate-zlib"
    ))]
    CompressionMethod::Deflated,
    #[cfg(feature = "bzip2")]
    CompressionMethod::Bzip2,
]
```

Reply:
Just was not sure is it nice enough :) Will fix it.

Reply:
As I thought:
```
rror: no rules expected the token `#`
   --> src/compression.rs:140:13
    |
140 |             #[cfg(any(
    |             ^ no rules expected this token in macro call

error: could not compile `zip` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed

```

Reply:
Nvm, fixed - my mistake

Reply:
Fixed